### PR TITLE
feat(sync): add manual trigger for conversation recovery mode

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Refactor the code to going in slowsync and recovery after 404 state to be able to trigger it manually for debugging purposes.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/RestartSlowSyncProcessForRecoveryUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    suspend operator fun invoke()
+}
+
+class RestartSlowSyncProcessForRecoveryUseCaseImpl internal constructor(
+    private val slowSyncRepository: SlowSyncRepository
+) : RestartSlowSyncProcessForRecoveryUseCase {
+    override suspend operator fun invoke() {
+        slowSyncRepository.clearLastSlowSyncCompletionInstant()
+        slowSyncRepository.setNeedsToRecoverMLSGroups(true)
+        slowSyncRepository.setNeedsToPersistHistoryLostMessage(true)
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/RestartSlowSyncProcessForRecoveryUseCase.kt:18]: Isn't this kinda by passing the whole 404 recovery logic. 

Can't we set `eventRepository.updateLastProcessedEventId()` to some bogus value and run the incremental sync. That should trigger a real 404 and the recovery logic. Also you don't need add these extra flags.

Comment (mchenani) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/RestartSlowSyncProcessForRecoveryUseCase.kt:18]: it's only the restarting section, the whole logic is happening the place we call this usecase.
The extra flag is also for the async process to insert the messages after the sync is done.

File: network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt [hunk 2]
```diff
@@ -110,15 +110,14 @@ internal fun provideBaseHttpClient(
 ) = HttpClient(engine) {
 
     if (NetworkLogger.isRequestLoggingEnabled) {
-        install(KaliumKtorCustomLogging) {
+        Logging {
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt:113]: Unrelated change which should we reverted?

File: network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt [hunk 3]
```diff
@@ -110,15 +110,14 @@ internal fun provideBaseHttpClient(
...
             logger = KaliumHttpLogger()
             level = LogLevel.ALL
         }
     }
 
     if (installCompression) {
         install(ContentEncoding) {
-            gzip()
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt:121]: Debug code?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ConversationsRecoveryManager.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+internal class ConversationsRecoveryManagerImpl(
+    private val incrementalSyncRepository: IncrementalSyncRepository,
+    private val addSystemMessageToAllConversationsUseCase: AddSystemMessageToAllConversationsUseCase,
+    private val slowSyncRepository: SlowSyncRepository,
+) : ConversationsRecoveryManager {
+
+    @Suppress("ComplexCondition")
+    override suspend fun invoke() {
+        // waitUntilIncrementalSyncIsDone
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ConversationsRecoveryManager.kt:20]:
```suggestion
        // wait until incremental sync is done
```

**Review by typfel — CHANGES_REQUESTED** [review]
👍 some debug code which needs to reverted and a nitpick.

File: network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt (hunk 2 above)
Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt:113]: oh! this one slipped out, sure will remove it. thanks

**Review by typfel — APPROVED** [review]
